src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const keyRemove of keysToRemove) {
+          delete state[keyRemove];
+        }
+
+        break;
+
+      case 'addProperties':
+        const { extraData } = action;
+
+        Object.assign(state, extraData);
```

Comment:
Consider using the spread operator instead of 'Object.assign' for better readability.